### PR TITLE
feat: Update issue severity api call + add short circuit logic

### PR DESCRIPTION
**Pull request by jennmueng**
Match the title input format with how the issue severity model was trained (merged exception type: exception value and fall back to title)

Add short circuit logic for fatal, info, and debug event levels.

### CODE REVIEW

File: tests/sentry/event_manager/test_severity.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def test_message_event_simple(
             {"logentry": {"message": "Dogs are great!"}},
         ]
         for case in cases:
-            manager = EventManager(make_event(**case))
+            manager = EventManager(make_event(level="error", **case))
```
Comment (isabellaenriquez) [inline tests/sentry/event_manager/test_severity.py:95]: nit: iirc `make_event` sets `level` to "error" by default, is there a reason this is made explicit here?

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -2127,29 +2127,30 @@ def _process_existing_aggregate(
...
 
 def _get_severity_score(event: Event) -> float | None:
-    # If the event is info-level or lower, auto-mark as low severity
-    if LOG_LEVELS_MAP[event.data["level"]] <= logging.INFO:
-        return 0
+    # Short circuit the severity value if we know the event is fatal or info/debug
+    level = event.data.get("level")
+    if LOG_LEVELS_MAP[level] == logging.FATAL:
+        return 1.0
+    if LOG_LEVELS_MAP[level] <= logging.INFO:
```
Comment (trillville) [inline src/sentry/event_manager.py:2134]: just confirming, this includes debug?

File: tests/sentry/event_manager/test_severity.py [hunk 3]
```diff
@@ -140,7 +140,37 @@ def test_usable_event_title(
...
+    def test_short_circuit_level(
+        self,
+        mock_urlopen: MagicMock,
+    ) -> None:
+        cases: list[tuple[str, float]] = [
+            ("fatal", 1.0),
+            ("info", 0.0),
+            ("debug", 0.0),
+            ("error", 0.1231),
+        ]
```
Comment (isabellaenriquez) [inline tests/sentry/event_manager/test_severity.py:161]: nice! `test_info_events_yield_zero_severity` can probably be removed?

**Review by isabellaenriquez — COMMENTED** [review]
just some nits :)

File: src/sentry/event_manager.py (hunk 2 above)
Comment (isabellaenriquez) [reply src/sentry/event_manager.py:2134]: it'd include debug and sample

Comment (jennmueng) [reply src/sentry/event_manager.py:2134]: Screenshot 2023-11-30 at 3 54 20 PM
yes

**Review by trillville — APPROVED** [review]

**Review by trillville — APPROVED** [review]
lgtm

**Review by isabellaenriquez — APPROVED** [review]
lgtm overall! nit: could add tests for when there isn't an `exception_type` and when there isn't an `exception_value` but nbd

**Pull request merged**